### PR TITLE
Allow binding guice objects as hk2 singletons via binding annotation.

### PR DESCRIPTION
**Pull request by jschlather**
Working on adding some functionality to time or log slow request/response filters. But the `InterceptionService` that jersey uses for AOP has to be bound as a singleton, which is not currently possible with the jersey2-guice bridge. I initially tried adding the annotation to the impl class, but there's not really anyway to access the impl class during binding. So adding a binding annotation that can be used on the interface class, was the only feasible way.

This approach seemed reasonable enough and could be used in the feature if needed.

### CODE REVIEW

**Comment by stevie400** [comment]
Should the annotation be on the class instead of in the binding key?

Tagging beaast + @jhaber: @kmclarnon @snommit-mit @Xcelled

**Comment by jschlather** [comment]
The use case for this is to bind `InterceptionService` which is an `hk2` class that we can't modify.

**Comment by stevie400** [comment]
Do you think there are a lot of places we'd use this? Since this is required to be bound as a singleton, would it make sense to just hardcode that somewhere in this module?

**Comment by jschlather** [comment]
Don't have a lot of other use cases right now, happy to hard code for this now. If that's preferable.

**Comment by stevie400** [comment]
I think that'd be easier to understand than having this new annotation across multiple repos. The change would definitely be simpler. If I'm wrong we can always change that.

File: jersey2-guice-impl/src/main/java/com/squarespace/jersey2/guice/GuiceBindingDescriptor.java [hunk 1]
```diff
@@ -43,8 +43,13 @@
   private final Binding<T> binding;
 
   public GuiceBindingDescriptor(Type type, Class<?> clazz,
-      Set<Annotation> qualifiers, Binding<T> binding) {
```
Comment (stevie400) [inline jersey2-guice-impl/src/main/java/com/squarespace/jersey2/guice/GuiceBindingDescriptor.java:46]: nit: can you please revert this whitespace change?

**Review by stevie400 — COMMENTED** [review]
LG

**Comment by stevie400** [comment]
still LGTM 😄 

Do you have merge permission?

**Comment by jschlather** [comment]
Yeah, doing some final testing of the bootstrap-rest feature.

**Comment by jschlather** [comment]
Ended up running into some strange issues with the stashes after binding the interception service as a singleton and using it. Which oddly enough worked fine in our unit tests. Going to close this out for now. I'll write a little more on the internal bootstrap PR.

**Pull request closed without merging**